src/request-parser/index.ts

```diff
@@ -0,0 +1,274 @@
+        ancestorFieldPath.push(getAlias(ancestorFieldNode) || getName(ancestorFieldNode) as string);
+      }
+    });
+
+    const fieldName = getAlias(fieldNode) || getName(fieldNode) as string;
+    ancestorFieldPath.push(fieldName);
+    const fieldPath = ancestorFieldPath.join(".");
+    const argumentsObjectMap = getArguments(fieldNode);
+    const directives = getDirectives(fieldNode);
+    let resourceValue: string | undefined;
+
+    if (argumentsObjectMap) {
+      if (argumentsObjectMap[resourceKey]) {
+        resourceValue = argumentsObjectMap[resourceKey];
+      } else if (variables && variables[resourceKey]) {
+        resourceValue = variables[resourceKey];
+      }
+    }
+
+    const { fieldTypeMap } = context;
+
+    fieldTypeMap.set(fieldPath, {
+      hasArguments: !!argumentsObjectMap,
+      hasDirectives: !!directives,
+      isEntity,
+      resourceValue,
+      typeName,
+    });
+  }
+
+  private _resourceKey: string = "id";
+  private _schema: GraphQLSchema;
+
+  constructor(schema: GraphQLSchema, resourceKey: string) {
+    this._resourceKey = resourceKey;
+    this._schema = schema;
+  }
+
+  public async updateRequest(
+    query: string,
+    opts: RequestOptions,
+    context: RequestContext,
+  ): Promise<{ ast: DocumentNode, query: string }> {
+    const _this = this;
+    const _query = opts.fragments ? RequestParser._concatFragments(query, opts.fragments) : query;
+    const typeInfo = new TypeInfo(this._schema);
+    let fragmentDefinitions: FragmentDefinitionNodeMap | undefined;
+
+    const ast = visit(parse(_query), {
+      enter(
+        node: ASTNode,
+        key: string | number,
+        parent: any,
+        path: Array<string | number>,
+        ancestors: any[],
+      ): ValueNode | undefined {
+        typeInfo.enter(node);
+        const kind = getKind(node);
+
+        if (kind === "Document") {
+          const documentNode = node as DocumentNode;
+          if (!opts.fragments || !hasFragmentDefinitions(documentNode)) return undefined;
+          fragmentDefinitions = getFragmentDefinitions(documentNode);
+          deleteFragmentDefinitions(documentNode);
+          return undefined;
+        }
+
+        if (kind === "Field" || kind === "InlineFragment") {
+          const fieldOrInlineFragmentNode = node as FieldNode | InlineFragmentNode;
+          const type = _this._getFieldOrInlineFragmentType(kind, fieldOrInlineFragmentNode, typeInfo);
+          if (!type) return undefined;
+
+          if (kind === "Field") {
+            const fieldNode = node as FieldNode;
+
+            if (fragmentDefinitions && hasFragmentSpreads(fieldNode)) {
+              setFragmentDefinitions(fragmentDefinitions, fieldNode);
+            }
+          }
+
+          if (!(type instanceof GraphQLObjectType) && !(type instanceof GraphQLInterfaceType)) return undefined;
+          const objectOrInterfaceType = type as GraphQLObjectType | GraphQLInterfaceType;

```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=typescript%3AS4325)